inception/inception-api-annotation-experimental/src/main/java/de/tudarmstadt/ukp/inception/experimental/api/AnnotationSystemAPIImpl.java

```diff
@@ -0,0 +1,418 @@
+            UserDao aUserDao, RepositoryProperties aRepositoryProperties,
+            AnnotationProcessAPI aAnnotationProcessAPI,
+            AnnotationSchemaService aAnnotationSchemaService,
+            AnnotationSystemAPIService aAnnotationSystemAPIService)
+    {
+        projectService = aProjectService;
+        documentService = aDocumentService;
+        userDao = aUserDao;
+        repositoryProperties = aRepositoryProperties;
+        annotationProcessAPI = aAnnotationProcessAPI;
+        annotationService = aAnnotationSchemaService;
+        annotationSystemAPIService = aAnnotationSystemAPIService;
+    }
+
+    @Override
+    public void handleDocument(ClientMessage aClientMessage) throws IOException {
+        try {
+            CAS cas;
+
+            DocumentMessage message = new DocumentMessage();
+
+            if (aClientMessage.getDocument() == 0L) {
+                // TODO receive random new document
+                cas = getCasForDocument(aClientMessage.getUsername(), aClientMessage.getProject(),
+                        41714);
+                message.setDocument(41714);
+            }
+            else {
+                cas = getCasForDocument(aClientMessage.getUsername(), aClientMessage.getProject(),
+                        aClientMessage.getDocument());
+
+                message.setDocument(
+                        toIntExact(documentService.getSourceDocument(aClientMessage.getProject(),
+                                aClientMessage.getDocument()).getId()));
+            }
+
+            message.setViewportText(getViewportText(aClientMessage, cas));
+
+            message.setAnnotations(
+                    filterAnnotations(getAnnotations(cas, aClientMessage.getProject()),
+                            aClientMessage.getViewport()));
+
+            annotationProcessAPI.handleSendDocumentRequest(message, aClientMessage.getClientName());
+        }
+        catch (Exception e) {
+            e.printStackTrace();
+            createErrorMessage(e.getMessage(), aClientMessage.getClientName());
+        }
+
+    }
+
+    @Override
+    public void handleViewport(ClientMessage aClientMessage) throws IOException
+    {
+        CAS cas = getCasForDocument(aClientMessage.getUsername(), aClientMessage.getProject(),
+                aClientMessage.getDocument());
+
+        ViewportMessage message = new ViewportMessage();
+
+        message.setViewportText(getViewportText(aClientMessage, cas));
+
+        message.setAnnotations(filterAnnotations(getAnnotations(cas, aClientMessage.getProject()),
+                aClientMessage.getViewport()));
+
+        annotationProcessAPI.handleSendViewportRequest(message, aClientMessage.getClientName());
+
+    }
+
+    @Override
+    public void handleSelectAnnotation(ClientMessage aClientMessage) throws IOException
+    {
+        CAS cas = getCasForDocument(aClientMessage.getUsername(), aClientMessage.getProject(),
```

Comment:
Why is the getCasForDocument method public and in the interface? Seems like something that should only be used internally.